### DISCUSSION TITLE
Audio-information handling

### DISCUSSION THREAD

**Discussion by Sohaib448506** [robotology/community#464] [Q&A] [answered]
Hi, as we can not access the Ears of ICUB. So i want to make the artificial ears in MATLAB. My purpose is that i give an audio file and tell him at what angle and distance the audio source is localized, then MATLAB will give me information what reached to left ear and what reached to right ear. I just want to know that information.
It's my start so i didn't want to localize the sound source. I just wanted to make a code which tells me what information reaches to left and right ear. Thanks waiting for your response.

**Comment by reafrancesco** [comment, marked as answer]
Hi, perhaps you can have a look to the Audio-Attention in Robotology.
This repository not only grabs audio from the left and right mic with different methods but it also does already localization with ITD for the iCub (with C++ code) in the YARP framework.
If you need further details please do not hesitate to contact me.

**Reply by pattacini** [reply]
This is the repo: https://github.com/robotology/audio-attention.

**Reply by Sohaib448506** [reply]
@reafrancesco I don't want to capture audio from mics. I want to make an MATLAB code which spread an audio file(provided with distance and angle with reference to a point) according to left and right ear. I want to capture information that reaches to left and right ear. Thanks

**Reply by pattacini** [reply]
@Sohaib448506 according to what @reafrancesco said

>This repository ... also does already localization with ITD

his suggestion seems to be still valid to me. Nonetheless, [`audio-attention`][1] doesn't appear to contain documentation, hence you may want to contact [Francesco][2] via email ✉

[1]: https://github.com/robotology/audio-attention
[2]: https://www.iit.it/people/francesco-rea

**Reply by reafrancesco** [reply]
Sure @Sohaib448506 feel free to contact me by email.
Further I have the documentation for the audio-attention ready. I am going to upload the documentation soon if it could be of any help.

**Comment by pattacini** [comment]
I've deleted #470 to favor this common thread to further discuss @Sohaib448506's request:

>Hi, i want to generate binural sound information reaching to our ears from any audio file in MATLAB? 
How can i do that?
